src/transformer/node-validator.ts

```
@@ -0,0 +1,89 @@
+/*!
+ * © 2025 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Node, NodeType } from 'prosemirror-model'
+import { findChildrenByType } from 'prosemirror-utils'
+
+export interface ValidationError {
+  code: string
+  nodeType?: string
+}
+
+export interface ValidationResult {
+  isValid: boolean
+  errors: ValidationError[]
+}
+
+export const VALIDATION_ERROR_CODES = {
+  MISSING_TITLE: 'MISSING_TITLE',
+  EMPTY_TITLE: 'EMPTY_TITLE',
+}
+
+const validateTitleNode = (nodes: Node[]): ValidationResult => {
```

I think we should just return the error list here. The "isValid" aggregation already happens at the top level.

---

I've updated the code, please let me know your thoughts.